### PR TITLE
UIDATIMP-516 Reference dropdowns. Replace previously-selected referen…

### PR DESCRIPTION
**Pull request by IvanKryzhanovskyi**
## Overview
In Instance, Holdings, and Item field mapping profiles, User should only be allowed to select one ref value from the dropdown list. If a second one is selected, it should replace the previously-selected one in the field mapping data field

## Approach
* Remove redundant property wrapperExplicitInsert
* Add updateDecoratorFieldValue handler
* Update test

## Ticket
[UIDATIMP-516](https://issues.folio.org/browse/UIDATIMP-516)

## Before
decorator_before

## After
decorator_after

### CODE REVIEW

File: src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.js [hunk 1]
```diff
@@ -37,12 +36,26 @@ export const withReferenceValues = memo(props => {
...
+    let newValue;
+    const containQuotedText = curVal.match(decoratorValueRegExp);
+
+    if (containQuotedText) {
+      newValue = `${curVal.replace(containQuotedText, `"${newVal}"`)}`;
+    } else {
+      newValue = `${curVal} "${newVal}"`;
+    }
+
+    return newValue;
```
Comment (VictorSoroka1) [inline src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.js:57]:
```suggestion
    return containQuotedText ? `${curVal.replace(containQuotedText, `"${newVal}"`)}` : `${curVal} "${newVal}"`;
```

File: src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.css [hunk 2]
```diff
@@ -6,11 +6,12 @@
   & .options-dropdown {
     display: flex;
     height: 100%;
-    margin: 25.5px 0 0 -1px;
+    margin: 25px 0 0 -1px;
```
Comment (mariia-aloshyna) [inline src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.css:9]: I see that these changes broke styles in some fields (see screenshot). I'm not able to explain why it happens, but `25.5px` solves this problem.
image

**Comment by mariia-aloshyna** [comment]
Update CHANGELOG, please.

File: src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.css (hunk 2 above)
Comment (IvanKryzhanovskyi) [reply src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.css]: It;s look like I managed issue with vertical decorator alignment in another way

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
 * Change placement of repeatable decorator in the Inventory field mapping screens (UIDATIMP-518)
 * Implement delete row functionality for the Field Mapping Profile details (UIDATIMP-498)
 * Implement re-order row functionality for the Field Mapping Profile details (UIDATIMP-496)
+* Update logic for reference dropdowns (UIDATIMP-516)
```
Comment (IhorKoba) [inline CHANGELOG.md:16]: Seems like it's a bug, not a story. So, could you please move description to bugfix section

Comment (IvanKryzhanovskyi) [reply CHANGELOG.md]: done

**Pull request merged**